src/orquestra/sdk/schema/ir.py

```diff
@@ -269,7 +270,22 @@ class ConstantNodePickle(BaseModel):
 ConstantNode = t.Union[ConstantNodeJSON, ConstantNodePickle]
 
 
-ArgumentId = t.Union[ArtifactNodeId, ConstantNodeId]
+class SecretNode(BaseModel):
+    """
+    A reference to a secret stored in an external secret/config service.
+    """
+
+    # Workflow-scope unique ID used to refer from task invocations
+    id: SecretNodeId
+
+    # Serialized value
+    secret_name: str
+
+    # Secret config
```

Comment:
**non-blocking,suggestion**: please state in the comment what's a situation where we expect this to be `None`

Reply:
wdyt? 543cc88